audit.py

```diff
@@ -715,5 +724,239 @@ def members(
+            ("Instance\nRef", {}),
+            ("Instance\nState", {}),
+        ]
+        super()._table_header(headers)
+
+    def _table_row(self, repo_name, alert):
+        if alert["dismissed_at"] is not None:
+            return
+
+        tool = alert.get("tool", {})
+        rule = alert.get("rule", {})
+        most_recent = alert.get("most_recent_instance", {})
+        row = [
+            repo_name,
+            alert.get("created_at"),
+            tool.get("name"),
+            tool.get("version"),
+            rule.get("name"),
+            rule.get("description"),
+            rule.get("security_severity_level"),
+            rule.get("severity"),
+            most_recent.get("ref"),
+            most_recent.get("state"),
+        ]
+        super()._table_row(*row)
+
+
+class RepoCodeScanAlertsHtmlReport(HtmlReportBase):
+
+    def _table_header(self):
+        EMPTY_COLUMN = ("", {})
+        headers_1 = [
+            ("Alert", dict(colspan=2, klass="column-group-header")),
+            EMPTY_COLUMN,
+            ("Tool", dict(colspan=2, klass="column-group-header")),
+            EMPTY_COLUMN,
+            ("Rule", dict(colspan=4, klass="column-group-header")),
+            EMPTY_COLUMN,
+            ("Instance", dict(colspan=4, klass="column-group-header")),
+        ]
+        headers_2 = [
+            ("Repository", {}),
+            ("Created", {}),
+            EMPTY_COLUMN,
+            ("Name", {}),
+            ("Version", {}),
+            EMPTY_COLUMN,
+            ("Name", {}),
+            ("Description", {}),
+            ("Level", {}),
+            ("Severity", {}),
+            EMPTY_COLUMN,
+            ("Ref", {}),
+            ("State", {}),
+        ]
+        super()._table_header(headers_1, headers_2)
+
+    def _table_row(self, repo_name, alert):
+        if alert["dismissed_at"] is not None:
+            return
+
+        tool = alert.get("tool", {})
+        rule = alert.get("rule", {})
+        most_recent = alert.get("most_recent_instance", {})
+        with self.tag("td"):
+            self.text(repo_name)
+        with self.tag("td"):
+            self.text(alert.get("created_at") or "")
+        self.doc.stag("td")
+        with self.tag("td"):
+            self.text(tool.get("name") or "")
+        with self.tag("td"):
+            self.text(tool.get("version") or "")
+        self.doc.stag("td")
+        with self.tag("td"):
+            self.text(rule.get("name") or "")
+        with self.tag("td"):
+            self.text(rule.get("description") or "")
+        with self.tag("td"):
+            self.text(rule.get("security_severity_level") or "")
+        with self.tag("td"):
+            self.text(rule.get("severity") or "")
+        self.doc.stag("td")
+        with self.tag("td"):
+            self.text(most_recent.get("ref") or "")
+        with self.tag("td"):
+            self.text(most_recent.get("state") or "")
+
+
+def convert_alert_instance(instance):
```

Comment:
Convert from what to what? Maybe add a docstring explaining what the conversion is doing?

Reply:
Added doc string